### DISCUSSION TITLE
C# session.Store Not working

### DISCUSSION THREAD

**Discussion by LuMiSxh** [ravendb/ravendb#12854] [Q&A] [answered]
Hello,
I have an issue with my C# code. I'm new to Raven but read the documentation and searched for solutions but wasnt able to find anything. 
My problem is, that i connect to my own node running on a Raspberry Pi via https and a certificate. I can initialize the Documentstore and start a session as well as save the changes the session made. The Problem itself is, that if i create an object and store it in the session, the `session.Store()` lines throws an exception: `System.AggregateException: "Failed to retrieve database topology from all known nodes.
[MY URL] -> The requested name is valid, but no data of the requested type was found. "` - this is the translated version, the original exception is in german. 

Code used:
```csharp
            using (IDocumentStore store = new DocumentStore
            {
                Urls = new[] { "MY URL" },
                Certificate = new X509Certificate2((string)Path.GetFullPath("Certificate.pfx")),
                Database = "Testing",
                Conventions = { }

            })
            {
                store.Initialize();


                using (IDocumentSession session = store.OpenSession())
                {
                    session.Advanced.WaitForIndexesAfterSaveChanges(
                    timeout: TimeSpan.FromSeconds(30),
                    throwOnTimeout: true);

                    session.Store(new Company { Name = "Company" , Contact = "Mail"});

                    session.SaveChanges();
                }
            }
```

The Company Class is the one from the example Models provided by Raven.

--- 
I would appreciate if anyone could help me solve this issue. Please keep in mind that this is my first time posting something on GitHub Discussion so my Markdown isnt perfect.

Greetings Luca

**Comment by ayende** [comment, marked as answer]
Is this what you have in your code?  
`   Urls = new[] { "MY URL" },`

If so, you need a real url here, something like: `http://live-test.ravendb.net/`

You can take the URL from the studio that you use to browse the server on the Pi.

**Reply by ppekrol** [reply]
The `The requested name is valid, but no data of the requested type was found.` is a DNS issue. Can you ping from the machine running your C# app that instance? Is the IP address resolved correctly? If yes, is that the address of that machine running RavenDB? I'm assuming that you can open Studio using that URL?

**Reply by LuMiSxh** [reply]
I don't know exactly what you mean with "ping" but I can tell you that the machine where the c# application is executed uses the Google DNS as well as the Raspberry pi. I can open the studio with the url and/or the ip + port without any problems on both machines. The IP + port I use for the RavenDB Server is `192.168.X.56:8081` (URL = `https://cla.Schmidt-LucaMichael.ravendb.Community:8081` --> The node I'm using is called cla)

**Reply by ayende** [reply]
From the Raspberry Pi as well as the machine that is running your C# code, please run.

`ping cla.Schmidt-LucaMichael.ravendb.Community`
`dig cla.Schmidt-LucaMichael.ravendb.Community`
`wget https://cla.Schmidt-LucaMichael.ravendb.Community:8081`

What are the results?

**Reply by ppekrol** [reply]
By `ping` I mean open a cmd/bash and execute `ping <exactly the url that you inserted into the Document Store>` - this works fine on your machine where the C# app is running? There are no connectivity issues between those 2 machines?

Does `curl -X GET https://cla.Schmidt-LucaMichael.ravendb.Community:8081/build/version` work? (should return that you do not have a certificate specified)

Can you also paste whole stack of the exception?

**Reply by LuMiSxh** [reply]
Commands on my Windows machine with the C# application :

- Ping: 
- - `Ping-Statistik für 192.168.178.56:
    Pakete: Gesendet = 4, Empfangen = 4, Verloren = 0
    (0% Verlust),
Ca. Zeitangaben in Millisek.:
    Minimum = 7ms, Maximum = 17ms, Mittelwert = 13ms`

- curl:
- - `{"BuildVersion":52012,"ProductVersion":"5.2","CommitHash":"999bd2d","FullVersion":"5.2.3"}`

Commands on my Raspberry Pi:

- Ping:
- - `64 bytes from 192.168.178.56 (192.168.178.56): icmp_seq=100 ttl=64 time=0.155 ms`

- Wget:
- - `--2021-10-15 17:15:02--  https://cla.schmidt-lucamichael.ravendb.community:8081/
Resolving cla.schmidt-lucamichael.ravendb.community (cla.schmidt-lucamichael.ravendb.community)... 192.168.178.56
Connecting to cla.schmidt-lucamichael.ravendb.community (cla.schmidt-lucamichael.ravendb.community)|192.168.178.56|:8081... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: /studio/index.html [following]
--2021-10-15 17:15:03--  https://cla.schmidt-lucamichael.ravendb.community:8081/studio/index.html
Reusing existing connection to cla.schmidt-lucamichael.ravendb.community:8081.
HTTP request sent, awaiting response... 403 Forbidden
2021-10-15 17:15:03 ERROR 403: Forbidden.
`

- Curl:
- - `{"Type":"InvalidAuth","Message":"This server requires client certificate for authentication, but none was provided by the client. Did you forget to install the certificate?\nYour certificate store may be cached by the browser. Create a new private browsing tab, which will not cache any certificates. (Ctrl+Shift+N in Chrome, Ctrl+Shift+P in Firefox)"}`

**Reply by LuMiSxh** [reply]
Uhmm... I dont know what happened but i went away from my PC to eat and when i came back just ago it started working again... I dont know why it didnt work before and why it works now but maybe you guys can give me some kind of awnser.

Thank you guys for helping

**Reply by ayende** [reply]
Might have been the DNS cache that took some time to update, about the only thing that I can think of.
